tests/v2/validation/rbac/rbac_cert_rotation_test.go

```diff
@@ -0,0 +1,151 @@
+	err := rotatecerts.RotateCertificates(rb.standardUserClient, rb.clusterName, rb.namespace)
+	switch role {
+	case roleOwner, restrictedAdmin:
+		require.NoError(rb.T(), err)
+
+	case roleMember, roleProjectOwner, roleProjectMember:
+		require.Error(rb.T(), err)
+		assert.Equal(rb.T(), "Resource type [provisioning.cattle.io.cluster] is not updatable", err.Error())
+	}
+
+	kubeProvisioningClient, err := rb.client.GetKubeAPIProvisioningClient()
+	require.NoError(rb.T(), err)
+
+	result, err := kubeProvisioningClient.Clusters(rb.namespace).Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + rb.clusterName,
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	})
+	require.NoError(rb.T(), err)
+
+	checkFunc := clusters.IsProvisioningClusterReady
+
+	err = wait.WatchWait(result, checkFunc)
+	assert.NoError(rb.T(), err)
+}
+
+func (rb *CertRotationRbacTestSuite) TestCertRotationRbac() {
+	clusterID, err := clusters.GetClusterIDByName(rb.client, rb.clusterName)
+	logrus.Infof("Cluster is %v", clusterID)
+	require.NoError(rb.T(), err)
+	if !(strings.Contains(clusterID, "c-m-")) {
+		rb.T().Skip("Skipping tests since cluster is not of type - k3s or RKE2")
+	}
+	tests := []struct {
+		name   string
+		role   string
+		member string
+	}{
+		{"Cluster Member", roleMember, standardUser},
+		{"Project Owner", roleProjectOwner, standardUser},
+		{"Project Member", roleProjectMember, standardUser},
+		{"Restricted Admin", restrictedAdmin, restrictedAdmin},
+		{"Cluster Owner", roleOwner, standardUser},
+	}
+	for _, tt := range tests {
+		rb.Run("Set up User with Cluster Role "+tt.name, func() {
+			newUser, err := createUser(rb.client, tt.member)
+			require.NoError(rb.T(), err)
+			rb.standardUser = newUser
+			rb.T().Logf("Created user: %v", rb.standardUser.Username)
+			rb.standardUserClient, err = rb.client.AsUser(newUser)
+			require.NoError(rb.T(), err)
+
+			subSession := rb.session.NewSession()
+			defer subSession.Cleanup()
+
+			createProjectAsAdmin, err := createProject(rb.client, rb.cluster.ID)
+			rb.adminProject = createProjectAsAdmin
+			require.NoError(rb.T(), err)
+		})
+
+		rb.Run("Adding user as "+tt.name+" to the downstream cluster.", func() {
+
+			if tt.member == standardUser {
+				if strings.Contains(tt.role, "project") {
+					err := users.AddProjectMember(rb.client, rb.adminProject, rb.standardUser, tt.role)
+					require.NoError(rb.T(), err)
+				} else {
+					err := users.AddClusterRoleToUser(rb.client, rb.cluster, rb.standardUser, tt.role)
+					require.NoError(rb.T(), err)
+				}
+			}
+
+			relogin, err := rb.standardUserClient.ReLogin()
+			require.NoError(rb.T(), err)
+			rb.standardUserClient = relogin
+		})
+
+		rb.T().Logf("Starting validations for %v", tt.role)
+
+		rb.Run("Test case - RotateCertificates of a cluster as a "+tt.name, func() {
+			rb.ValidateCertRotationCluster(tt.role)
+		})
+
+	}
+}
+
+func TestCertRotationRbacTestSuite(t *testing.T) {
+	logrus.Infof("Beginning testing")
```

Comment:
can we remove the log here? as we have "Starting validations" log added in the TestCertRotationRbac?

Reply:
agreed